### ISSUE TITLE
Futures are not Send

### ISSUE THREAD

**Issue by kebabtent** [ramsayleung/rspotify#236]
I am using the latest commit in the master branch. The futures produced by the client are not `Send`, meaning the client cannot be used inside a spawned task. Is there any way around it or are there plans to resolve it?

Minimal example:
```rust
#[tokio::main]
async fn main() {
    tokio::spawn(spotify());
}

async fn spotify() {
    let creds = rspotify::Credentials::from_env().unwrap();
    let mut spotify = rspotify::ClientCredsSpotify::new(creds);
    spotify.request_token().await.unwrap();
}
```

**Comment by marioortizmanero** [comment]
Hmm. The futures are `Send` if `ureq` is being used, but not for `reqwest`. I'll investigate more, but it doesn't seem to be `async_trait`'s fault.

**Comment by marioortizmanero** [comment]
Ok, this seems to be a side effect of using `#[async_trait(?Send)]` for the `BaseClient` trait.

**Comment by marioortizmanero** [comment]
This is fixed by #237
